### PR TITLE
populating colours from hexa-decimal value's

### PR DESCRIPTION
**Pull request by PurushYM**
- statusBarcolorFromHex
- closeButtonColorFromHex

### CODE REVIEW

File: YmChat/YmChat.apple.cs [hunk 1]
```diff
@@ -107,5 +107,61 @@ public void setVersion(int version)
...
+
+            switch (colorString.Length)
+            {
+                case 3: // #RGB
+                    {
+                        red = Convert.ToInt32(string.Format("{0}{0}", colorString.Substring(0, 1)), 16) / 255f;
+                        green = Convert.ToInt32(string.Format("{0}{0}", colorString.Substring(1, 1)), 16) / 255f;
+                        blue = Convert.ToInt32(string.Format("{0}{0}", colorString.Substring(2, 1)), 16) / 255f;
+                        return UIColor.FromRGB(red, green, blue);
+                    }
```
Comment (kaunteya) [inline YmChat/YmChat.apple.cs:151]: We can consider only 6 letter hex format #123456. Let's skip 3 character format

**Review by kaunteya — CHANGES_REQUESTED** [review]

File: YmChat/YmChat.apple.cs (hunk 1 above)
Comment (PurushYM) [reply YmChat/YmChat.apple.cs:151]: ok

**Review by kaunteya — APPROVED** [review]
iOS part looks good

File: YmChat.nuspec [hunk 2]
```diff
@@ -2,13 +2,13 @@
...
     <id>YmChat</id>
-    <version>1.2.3</version>
+    <version>1.4.0</version>
     <title>The YmChat Plugin for Xamarin</title>
     <authors>Purushottam</authors>
     <requireLicenseAcceptance>false</requireLicenseAcceptance>
     <projectUrl>https://github.com/yellowmessenger/ymchat-xamarin</projectUrl>
     <description>The Nuget Package for integrating yellow.ai chat bot into xamarin forms application</description>
-    <releaseNotes>Supports Android and iOS platforms</releaseNotes>
+    <releaseNotes>Added setCustomLoaderURL API (for setting custom loading image for chatbot)</releaseNotes>
```
Comment (sukrit-ym) [inline YmChat.nuspec:11]: update the message with StatusBarColor and CloseButtonColor API

File: YmChat/YmChat.apple.cs [hunk 3]
```diff
@@ -94,18 +94,58 @@ public void onBotClose(Action callback)
...
+
+                default:
+                    throw new ArgumentOutOfRangeException(string.Format("Invalid color value {0}. It should be a hex value of the form #RRGGBB", hexValue));
+
+            }
+        }
+    }
+
```
Comment (sukrit-ym) [inline YmChat/YmChat.apple.cs:150]: Why is this required?

**Review by sukrit-ym — CHANGES_REQUESTED** [review]

File: YmChat/YmChat.apple.cs (hunk 3 above)
Comment (PurushYM) [reply YmChat/YmChat.apple.cs:150]: For changing Hex color to iOS UI color

File: YmChat.nuspec [hunk 4]
```diff
@@ -2,13 +2,13 @@
...
     <id>YmChat</id>
-    <version>1.2.3</version>
+    <version>1.4.0</version>
     <title>The YmChat Plugin for Xamarin</title>
     <authors>Purushottam</authors>
     <requireLicenseAcceptance>false</requireLicenseAcceptance>
     <projectUrl>https://github.com/yellowmessenger/ymchat-xamarin</projectUrl>
     <description>The Nuget Package for integrating yellow.ai chat bot into xamarin forms application</description>
-    <releaseNotes>Supports Android and iOS platforms</releaseNotes>
+    <releaseNotes>Added API's for setting color tp statusbar and clpseButton </releaseNotes>
```
Comment (sukrit-ym) [inline YmChat.nuspec:11]: Please correct the spellings

Comment (sukrit-ym) [reply YmChat.nuspec:11]: Message - " Added API's for custom bot loader using a url, setting color of status bar and close button with hexadecimal color value."

**Review by sukrit-ym — APPROVED** [review]

**Pull request merged**